presentation-api/controlling-ua/PresentationRequest_onconnectionavailable-manual.html

```diff
@@ -0,0 +1,54 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>PresentationRequest.onconnectionavailable (manual test)</title>
+<link rel="author" title="Tomoyuki Shimizu" href="https://github.com/tomoyukilabs">
+<link rel="help" href="https://w3c.github.io/presentation-api/#starting-a-presentation">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+
+<p>Click the button below and select the available presentation display, to start the manual test.</p>
+<button id="presentBtn" onclick="startPresentation()">Start Presentation Test</button>
+
+
+<script>
+    // ----------
+    // DOM Object
+    // ----------
+    var presentBtn = document.getElementById("presentBtn");
+
+    // ------------
+    // Request init
+    // ------------
+    var validUnixDate = new Date().getTime() + String(Math.floor(Math.random() * 1e5)),
+            validPresURL = '../receiving-ua/idlharness.html#__castAppId__=2334D33A/__castClientId__=' + validUnixDate,
+            request = new PresentationRequest(validPresURL);
+
+    // --------------------------------------------------------------------------
+    // Start New PresentationRequest.onconnectionavailable Test (success) - begin
+    // --------------------------------------------------------------------------
+    var startPresentation = function () {
+        async_test(function (t) {
+            request.onconnectionavailable = t.step_func_done(function (evt) {
+                var connection = evt.connection;
+
+                // Check the initial state of a presentation connection
+                assert_equals(connection.state, 'connecting', 'The initial state of the presentation connection is "connecting".');
+
+                // Check, if the connection ID is set
+                assert_true(!!connection.id, 'The connection ID is set.');
+
+                // Check the type of the connection.id
+                assert_true(typeof connection.id === 'string', 'The connection ID is a string.');
+
+                // Check the instance of the connection
+                assert_true(connection instanceof PresentationConnection, 'The connection is an instance of PresentationConnection.');
+            });
+
+            request.start();
```

Comment:
Another problem here is that there's no code to handle the case where the call to `start` fails. The Promise will be rejected but that will go unnoticed. To fix this, `promise_test` should be used instead of `async_test`, and the Promise should be returned as in:

```js
promise_test(function (t) {
  request.onconnectionavailable = [...];
  return request.start();
});
```